modules/gentemplates/openapi-normalizer/src/test/java/com/reprezen/genflow/openapi/normalizer/GenTemplateTest.xtend

```diff
@@ -0,0 +1,27 @@
+package com.reprezen.genflow.openapi.normalizer
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.google.common.io.Resources
+import com.reprezen.genflow.api.target.GenTargetUtils
+import java.io.File
+import org.junit.Test
+
+import static org.junit.Assert.*
+
+class GenTemplateTest {
+
+	@Test
+	def void testGenTemplate() {
+		val gt = GenTargetUtils.load(
+			new File(Resources.getResource(GenTemplateTest, "/normalizer_path_or_components.gen").toURI))
+
+		val traces = gt.execute
+		val output = new File(new File(traces.baseDirectory, "generated"), "Swagger Petstore.json")
+		
+		val tree = new ObjectMapper().readTree(output)
+
+		assertFalse(tree.get("paths").isEmpty)
+		assertFalse(tree.get("components").isEmpty)
```

Comment:
I think one of these two assertions should always fail with PATH_OR_COMPONENTS. 

The normalized output should contain paths XOR components, not both.

Reply:
Yes in this example the components are used in paths, so they are retained.
It would need more test cases.